src/backend/distributed/utils/metadata_cache.c

```diff
@@ -89,6 +92,43 @@ IsDistributedTable(Oid relationId)
 
 
 /*
+ * HasUninitializedShardInterval returns true if the distributed relation has
+ * at least one shard interval whose min/max value has not been set.
+ */
+bool
+HasUninitializedShardInterval(Oid relationId)
```

Comment:
It seems better to store this as an attribute to DistTableCacheEntry. E.g. DistTableCacheEntry->shardIntervalHasUninitialized (and then DistTableCacheEntry->shardIntervalIsUniform).